### PR TITLE
chores: QGB keepers specific errors

### LINKED ISSUE #1390: QGB keepers specific errors
As discussed in https://github.com/celestiaorg/celestia-app/pull/1389#discussion_r1108437093, currently, we're creating errors on the fly inside the QGB keepers using the `fmt.Errorf` or  similar. We could improve that by creating specific errors.

### PR DESCRIPTION
**Pull request by phamminh0811**
## Overview

closes #1390, this pr add some specific errors on QGB keepers data commitment and make sure previous tests success

## Checklist

- [ ] New and updated code has appropriate documentation
- [x] New and updated code has new updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: testutil/testnode/full_node.go [hunk 1]
```diff
@@ -214,11 +215,34 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (accounts []string, c
 		t.Log("tearing down testnode")
 		require.NoError(t, stopNode())
 		require.NoError(t, cleanupGRPC())
+		require.NoError(t, removeDir(path.Join([]string{tmCfg.RootDir, "config"}...)))
```
Comment (SweeXordious) [inline testutil/testnode/full_node.go:218]: [blocking]
Could you revert this change please? This is introduced in another PR

File: testutil/testnode/full_node.go [hunk 2]
```diff
@@ -214,11 +215,34 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (accounts []string, c
...
+			return err
+		}
+	}
+	err = os.RemoveAll(rootDir)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
```
Comment (SweeXordious) [inline testutil/testnode/full_node.go:245]: [blocking]
Similar, could you please revert?

File: testutil/testnode/rpc_client.go [hunk 3]
```diff
@@ -36,7 +34,7 @@ func StartNode(tmNode *node.Node, cctx Context) (Context, func() error, error) {
 			return err
 		}
 		tmNode.Wait()
-		return removeDir(path.Join([]string{cctx.HomeDir, "config"}...))
+		return nil
```
Comment (SweeXordious) [inline testutil/testnode/rpc_client.go:37]: [blocking]
same

File: testutil/testnode/rpc_client.go [hunk 4]
```diff
@@ -76,22 +74,3 @@ func StartGRPCServer(app srvtypes.Application, appCfg *srvconfig.Config, cctx Co
 func DefaultAppConfig() *srvconfig.Config {
 	return srvconfig.DefaultConfig()
 }
-
-// removeDir removes the directory `rootDir`.
```
Comment (SweeXordious) [inline testutil/testnode/rpc_client.go:80]: [blocking]
same

File: x/qgb/types/errors.go [hunk 5]
```diff
@@ -15,4 +15,6 @@ var (
 	ErrUnmarshalllAttestation                = sdkerrors.Register(ModuleName, 26, "couldn't unmarshall attestation from store")
 	ErrNonceHigherThanLatestAttestationNonce = sdkerrors.Register(ModuleName, 27, "the provided nonce is higher than the latest attestation nonce")
 	ErrNoValsetBeforeNonceOne                = sdkerrors.Register(ModuleName, 28, "there is no valset before attestation nonce 1")
+	ErrDataCommitmentNotGenerated			 = sdkerrors.Register(ModuleName, 29, "there is no data commitment has been generated")
```
Comment (SweeXordious) [inline x/qgb/types/errors.go:18]: [blocking]
Could you please run ` $ gofumpt -l -w .` to correctly format this?

Comment (SweeXordious) [inline x/qgb/types/errors.go:18]:
```suggestion
	ErrDataCommitmentNotGenerated			 = sdkerrors.Register(ModuleName, 29, "no data commitment has been generated for the provided height")
```

File: x/qgb/keeper/keeper_data_commitment.go [hunk 6]
```diff
@@ -73,13 +77,13 @@ func (k Keeper) GetLastDataCommitment(ctx sdk.Context) (types.DataCommitment, er
...
+			return types.DataCommitment{}, sdkerrors.Wrapf(types.ErrAttestationNotFound, fmt.Sprintf("couldn't find attestation with nonce %d", latestNonce-i))
 		}
 		dcc, ok := att.(*types.DataCommitment)
 		if !ok {
 			continue
 		}
 		return *dcc, nil
 	}
-	return types.DataCommitment{}, fmt.Errorf("last data commitment not found")
+	return types.DataCommitment{}, sdkerrors.Wrap(types.ErrDataCommitmentNotFound, "last data commitment not found")
```
Comment (SweeXordious) [inline x/qgb/keeper/keeper_data_commitment.go:88]:
```suggestion
	return types.DataCommitment{}, types.ErrDataCommitmentNotFound
```

I think we don't need to wrap if we're already providing the error that has the same message

File: x/qgb/keeper/keeper_data_commitment.go [hunk 7]
```diff
@@ -73,13 +77,13 @@ func (k Keeper) GetLastDataCommitment(ctx sdk.Context) (types.DataCommitment, er
 			return types.DataCommitment{}, err
 		}
 		if !found {
-			return types.DataCommitment{}, fmt.Errorf("couldn't find attestation with nonce %d", latestNonce-i)
+			return types.DataCommitment{}, sdkerrors.Wrapf(types.ErrAttestationNotFound, fmt.Sprintf("couldn't find attestation with nonce %d", latestNonce-i))
```
Comment (SweeXordious) [inline x/qgb/keeper/keeper_data_commitment.go:80]:
```suggestion
			return types.DataCommitment{}, sdkerrors.Wrapf(types.ErrAttestationNotFound, fmt.Sprintf("nonce %d", latestNonce-i))
```

I think no need to repeat the message

File: x/qgb/keeper/keeper_data_commitment.go [hunk 8]
```diff
@@ -51,7 +55,7 @@ func (k Keeper) GetDataCommitmentForHeight(ctx sdk.Context, height uint64) (type
 			return types.DataCommitment{}, err
 		}
 		if !found {
-			return types.DataCommitment{}, fmt.Errorf("couldn't find attestation with nonce %d", i)
+			return types.DataCommitment{}, sdkerrors.Wrap(types.ErrAttestationNotFound, fmt.Sprintf("couldn't find attestation with nonce %d", i))
```
Comment (SweeXordious) [inline x/qgb/keeper/keeper_data_commitment.go:58]:
```suggestion
			return types.DataCommitment{}, sdkerrors.Wrap(types.ErrAttestationNotFound, fmt.Sprintf("nonce %d", i))
```

File: x/qgb/keeper/keeper_data_commitment.go [hunk 9]
```diff
@@ -37,10 +38,13 @@ func (k Keeper) GetDataCommitmentForHeight(ctx sdk.Context, height uint64) (type
...
 	}
 	if lastDC.EndBlock < height {
-		return types.DataCommitment{}, fmt.Errorf(
-			"no data commitment has been generated for the provided height. Last height %d < %d",
-			lastDC.EndBlock,
-			height,
+		return types.DataCommitment{}, sdkerrors.Wrap(
+			types.ErrDataCommitmentNotGenerated,
+			fmt.Sprintf(
+				"no data commitment has been generated for the provided height. Last height %d < %d",
```
Comment (SweeXordious) [inline x/qgb/keeper/keeper_data_commitment.go:44]:
```suggestion
				"Last height %d < %d",
```

**Review by SweeXordious — CHANGES_REQUESTED** [review]
Thanks a lot for handling this :+1: :+1: :rocket: 

Left some blocking comments, ping me to re-review when they're addressed

**Comment by phamminh0811** [comment]
> Thanks a lot for handling this +1 +1 rocket
> 
> Left some blocking comments, ping me to re-review when they're addressed

it's all resolved, please check

**Review by SweeXordious — APPROVED** [review]
:rocket: :rocket:

**Review by evan-forbes — APPROVED** [review]
thanks @phamminh0811

**Pull request merged**